src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -299,9 +300,15 @@ typeClassMemberToDictionaryAccessor mn name args (TypeDeclaration (TypeDeclarati
   in ValueDecl sa ident Private []
     [MkUnguarded (
      TypedValue False (Abs (VarBinder ss dictIdent) (Case [Var ss $ Qualified Nothing dictIdent] [CaseAlternative [ctor] [MkUnguarded acsr]])) $
-       moveQuantifiersToFront (quantify (srcConstrainedType (srcConstraint className [] (map (srcTypeVar . fst) args) Nothing) ty))
+       makeVta (moveQuantifiersToFront (quantify (srcConstrainedType (srcConstraint className [] (map (srcTypeVar . fst) args) Nothing) ty)))
     )]
-typeClassMemberToDictionaryAccessor _ _ _ _ = internalError "Invalid declaration in type class definition"
+  where
+  vtaVars = S.fromList $ map fst $ filter ((== IsVtaTypeVar) . snd) $ zip (map fst args) vtas
+  makeVta = everywhereOnTypes go
+    where
+    go (ForAll a b c d e _) | b `S.member` vtaVars = ForAll a b c d e IsVtaTypeVar
+    go t = t
```

Comment:
I don't think this rewrite is handling scope (ie, it's assuming all forall binder names are unique). Is it possible for shadowed binders to potentially mess with it?

Reply:
It's highly likely, yeah. I'll factor this out as a helper function.